schema/hardware-metadata.schema.json

```diff
@@ -0,0 +1,261 @@
+{
+  "$id": "https://zmkfirmware.dev/zmk.metadata.json",
+  "title": "HardwareMetadata",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "oneOf": [
+    {
+      "$ref": "#/$defs/board"
+    },
+    {
+      "$ref": "#/$defs/shield"
+    },
+    {
+      "$ref": "#/$defs/interconnect"
+    }
+  ],
+  "$defs": {
+    "id": {
+      "type": "string",
+      "pattern": "^[a-z0-9_]+$"
```

Comment:
Is it okay for an ID to start with a number? (Not sure if these IDs need to match C identifier rules.)

Reply:
These do *not* need to be C identifiers, and we should be a bit more flexible here, for when someone has a keyboard named funky things like _33ble (Looking at you, @tominabox1 )

Reply:
That works. Just wanted to make sure we didn't need this to be `[a-z_][a-z0-9_]*`